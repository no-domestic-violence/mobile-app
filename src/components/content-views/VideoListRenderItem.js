import React from 'react';
import { View, Image, TouchableOpacity, StyleSheet, Text } from 'react-native';
import { faPlay } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';

const VideoListRenderItem = ({ item, navigation }) => {
  return (
    <TouchableOpacity
      style={styles.itemContainer}
      onPress={() => navigation.navigate('Video Page', { params: item })}>
      <Image source={item.image} style={styles.itemImage} />
      <FontAwesomeIcon
        icon={faPlay}
        size={38}
        color='#415889'
        style={styles.itemIcon}
      />
      <Text style={styles.title}>{item.title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  itemContainer: {
    margin: 20,
    alignItems: 'center',
    borderBottomWidth: 1,
    paddingBottom: 20,
    borderColor: 'grey',
  },
  itemIcon: {
    position: 'absolute',
    top: '45%',
    left: '45%',
  },
  itemImage: {
    height: 200,
    width: '100%',
    marginBottom: 20,
  },
  title: {
    fontSize: 16,
    fontWeight: '500',
  },
});

export default VideoListRenderItem;
