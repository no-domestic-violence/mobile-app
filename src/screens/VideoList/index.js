import React, {useState} from 'react';
import {View, Text, Button, FlatList} from 'react-native';
import {videoData} from '../../mock/video'
import VideoListRenderItem from '../../components/content-views/VideoListRenderItem';

export default function VideoList({navigation}) {
  const [watchList, setWatchList] = useState(videoData);
 const onPressItem = () => {
  navigation.navigate('Video Page')
 }
  return (
    <View>
        <FlatList
          data={watchList}
          renderItem={({ item }) => <VideoListRenderItem item={item} onPress={onPressItem}/>}
        />
        <Button
            title="Go to video"
            onPress={() =>
              navigation.navigate('Video Page')
            }
          />
    </View>
  );
}
