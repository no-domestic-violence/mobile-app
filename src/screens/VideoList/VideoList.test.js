import React from 'react';
import { render } from '@testing-library/react-native';
import VideoListRenderItem from '_components/content-views/VideoListRenderItem';
import VideoList from './index';

export const mockDataVideos = [
  {
    image: 'background1',
    id: 1,
    title: '7 Early Signs of A Toxic Relationship',
    url: 'https://www.youtube.com/embed/HDNMvuQrcGk',
  },
  {
    image: 'background2',
    id: 2,
    title: 'Why domestic violence victims do not leave | Leslie Morgan Steiner',
    url: 'https://www.youtube.com/embed/V1yW5IsnSjo',
  },
  {
    image: 'background3',
    id: 3,
    title:
      'I broke my silence: My story of domestic violence | Emma Murphy | TEDxUniversityofNicosia',
    url: 'https://www.youtube.com/embed/frFEdN7aMh8',
  },
  {
    image: 'background4',
    id: 4,
    title: 'Dangerous Love (Domestic Abuse Documentary) | Real Stories',
    url: 'https://www.youtube.com/embed/0yIAY2tbnOE',
  },
  {
    image: 'background5',
    id: 5,
    title: 'The difference between healthy and unhealthy love | Katie Hood',
    url: 'https://www.youtube.com/embed/ON4iy8hq2hM',
  },
];

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

describe('Testing', () => {
  it('should match snapshot', () => {
    const result = render(<VideoList />).toJSON();
    expect(result).toMatchSnapshot();
  });
/* TODO: continue later
  it('should display the order as a <ListItem />', () => {
    const element = wrapper.instance().renderItem(item);
    expect(element.type).toBe(ListItem);
    expect(element.props).toEqual({
      title: 'Chris Jackson',
      subtitle: 'Vice Chairman',
      leftAvatar: {
        source: {
          uri:
            'https://s3.amazonaws.com/uifaces/faces/twitter/adhamdannaway/128.jpg',
        },
        title: 'C',
      },
    });
  });
  */
});