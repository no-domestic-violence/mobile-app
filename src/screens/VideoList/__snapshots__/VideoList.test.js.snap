// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing should match snapshot 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "image": Object {
            "testUri": "../../../src/assets/backgrounds/video1.png",
          },
          "title": "7 Early Signs of A Toxic Relationship",
          "url": "https://www.youtube.com/embed/HDNMvuQrcGk",
        },
        Object {
          "id": 2,
          "image": Object {
            "testUri": "../../../src/assets/backgrounds/video2.png",
          },
          "title": "Why domestic violence victims do not leave | Leslie Morgan Steiner",
          "url": "https://www.youtube.com/embed/V1yW5IsnSjo",
        },
        Object {
          "id": 3,
          "image": Object {
            "testUri": "../../../src/assets/backgrounds/video3.png",
          },
          "title": "I broke my silence: My story of domestic violence | Emma Murphy | TEDxUniversityofNicosia",
          "url": "https://www.youtube.com/embed/frFEdN7aMh8",
        },
        Object {
          "id": 4,
          "image": Object {
            "testUri": "../../../src/assets/backgrounds/video4.png",
          },
          "title": "Dangerous Love (Domestic Abuse Documentary) | Real Stories",
          "url": "https://www.youtube.com/embed/0yIAY2tbnOE",
        },
        Object {
          "id": 5,
          "image": Object {
            "testUri": "../../../src/assets/backgrounds/video5.png",
          },
          "title": "The difference between healthy and unhealthy love | Katie Hood",
          "url": "https://www.youtube.com/embed/ON4iy8hq2hM",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "grey",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/backgrounds/video1.png",
              }
            }
            style={
              Object {
                "height": 200,
                "marginBottom": 20,
                "width": "100%",
              }
            }
          />
          <
            color="#415889"
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f04b",
                  "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
                ],
                "iconName": "play",
                "prefix": "fas",
              }
            }
            size={38}
            style={
              Object {
                "left": "45%",
                "position": "absolute",
                "top": "45%",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            7 Early Signs of A Toxic Relationship
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "grey",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/backgrounds/video2.png",
              }
            }
            style={
              Object {
                "height": 200,
                "marginBottom": 20,
                "width": "100%",
              }
            }
          />
          <
            color="#415889"
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f04b",
                  "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
                ],
                "iconName": "play",
                "prefix": "fas",
              }
            }
            size={38}
            style={
              Object {
                "left": "45%",
                "position": "absolute",
                "top": "45%",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Why domestic violence victims do not leave | Leslie Morgan Steiner
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "grey",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/backgrounds/video3.png",
              }
            }
            style={
              Object {
                "height": 200,
                "marginBottom": 20,
                "width": "100%",
              }
            }
          />
          <
            color="#415889"
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f04b",
                  "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
                ],
                "iconName": "play",
                "prefix": "fas",
              }
            }
            size={38}
            style={
              Object {
                "left": "45%",
                "position": "absolute",
                "top": "45%",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            I broke my silence: My story of domestic violence | Emma Murphy | TEDxUniversityofNicosia
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "grey",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/backgrounds/video4.png",
              }
            }
            style={
              Object {
                "height": 200,
                "marginBottom": 20,
                "width": "100%",
              }
            }
          />
          <
            color="#415889"
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f04b",
                  "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
                ],
                "iconName": "play",
                "prefix": "fas",
              }
            }
            size={38}
            style={
              Object {
                "left": "45%",
                "position": "absolute",
                "top": "45%",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Dangerous Love (Domestic Abuse Documentary) | Real Stories
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "grey",
              "margin": 20,
              "opacity": 1,
              "paddingBottom": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/backgrounds/video5.png",
              }
            }
            style={
              Object {
                "height": 200,
                "marginBottom": 20,
                "width": "100%",
              }
            }
          />
          <
            color="#415889"
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f04b",
                  "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
                ],
                "iconName": "play",
                "prefix": "fas",
              }
            }
            size={38}
            style={
              Object {
                "left": "45%",
                "position": "absolute",
                "top": "45%",
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            The difference between healthy and unhealthy love | Katie Hood
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
